enhancements/security/scc-csi-volumes.md

```diff
@@ -0,0 +1,234 @@
+A service account/user can either mount a `csi` volume, or it cannot.
+This proposal will enhance SCCs to control which CSI drivers are allowed.
+
+## Motivation
+
+Inline CSI volumes are provided via CSI drivers that can be added to OpenShift.
+When a pod mounts an inline `csi` volume, it must specify the driver that will provide the underlying content.
+Cluster admins will need a means to identify which drivers are considered safe for use, including in restricted namespaces.
+
+Ephemeral CSI drivers are expected to be used in the following use cases:
+
+1. Mounting shared simple content access certificates used to install RHEL content.
+2. Accessing [sealed secrets](https://secrets-store-csi-driver.sigs.k8s.io/) stored in Vault, Azure Key Vault, or Google Secret Manager.
+
+### Goals
+
+* Allow cluster admins to control which ephemeral CSI drivers can be used to mount `csi` volumes.
+
+### Non-Goals
+
+* Allow `csi` volumes to be mounted in build pods.
+* Enable SCCs to limit [generic ephemeral volumes](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes)
+
+## Proposal
+
+### User Stories
+
+As an OpenShift cluster admin
+I want to control which CSI drivers can be used to mount csi volumes
+So that my workloads only use trusted CSI drivers to access content
+
+As an application developer
+I want to use the secret store CSI driver to mount csi volumes
+So that I can access sealed secrets in my workloads
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Safe for Consumption Annotation
+
+CSI drivers that provide inline CSI volumes must mark themselves as safe for consumption.
+To mark a CSI driver safe, admins or an operator will need to add the `security.openshift.io/allow-csi-volumes-restricted` annotation to the driver's `CSIDriver` object:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: CSIDriver
+annotations:
+  security.openshift.io/allow-csi-volumes-restricted: "true"
+  ...
+metadata:
+  name: mycsi.dev
+  ...
+```
+
+#### Pod admission
+
+Pod that mount `csi` volumes cannot be admitted unless the chosen SCC allows `csi` volumes.
+In addition, the driver for the volume mount must have the `security.openshift.io/allow-csi-volumes-restricted` annotation.
+
+#### Allow CSI volumes in restricted namespaces
+
+Ephemeral CSI drivers will be promoted to the list of "safe" volume mounts, and be considered equivalent to a Secret or ConfigMap from a safety/restriction perspective.
+CSI volumes will be added to the list of allowed volume types in the `restricted` SCC.
+All default SCCs installed by OpenShift with less restrictions will likewise allow `csi` volumes.
+When SCCs are ordered, the `csi` volume type will count as a "trivial" volume mount.
+
+### Risks and Mitigations
+
+**Risk:**
+Default security context constraints will allow any CSI driver to be used.
+
+*Mitigation:*
+Only the CSI drivers that are marked safe by an operator or admin will be allowed to mount `csi` volumes.
```

Comment:
what does this mean for the existing SCCs that allow all CSI drivers today?  Are we going to potentially lock down users from being able to use CSI drivers they used to be able to use, because they aren't explicitly marked "safe"?